### PR TITLE
ENH: adding reverse_streamlines

### PR DESCRIPTION
**Pull request by EmmaRenauld**
## Description

Changed this function to use with SFT. Now deals with data_per_point and _per_streamline. 

## Testing data and script

Use your favorite SFT. To see data_per_point, use trk containing some information. Ex, with MI-Brain, color it and then save it. Example of use:
```
from dwi_ml.data.processing.streamlines.data_augmentation import reverse_streamlines
from dipy.io.streamline import (load_tractogram, save_trk)

sft = load_tractogram(YOUR_BUNDLE, YOUR_REF)
reverse_ids = np.arange(len(sft.streamlines)/2)
sft2 = reverse_streamlines(sft, reverse_ids)
save_trk(sft2, 'test_reverse_streamlines.trk')
```
It is a little difficult to visualize results in a vizualizing program such as MI-Brain. Here is how you could check results. Compare outputs for:

```
sft.streamlines[0][0]
sft2.streamlines[0][0]  # Not the same
sft2.streamlines[0][-1] # The same

# Ex, with a colored dataset:
sft.data_per_point['color_x'][0][0:5]  # Five first points of first streamline
sft2.data_per_point['color_x'][0][0:5] # Not the same
sft2.data_per_point['color_x'][0][-5:] # The same, flipped
```

## Have you
- [x ] Added a description of the content of this PR above
- [ ] Followed [proper commit message formatting](https://chris.beams.io/posts/git-commit/)
- [x ] Added data and a script to test this PR
- [x ] Made sure that PEP8 issues are resolved
- [x ] Tested the code yourself right before pushing
- [ ] Added unit tests to test the code you implemented

### CODE REVIEW

File: dwi_ml/data/processing/streamlines/data_augmentation.py [hunk 1]
```diff
@@ -103,5 +103,33 @@ def cut_random_streamlines(
...
     return output_streamlines
 
-def flip_streamlines():
-    #TODO!!
\ No newline at end of file
+
+def reverse_streamlines(sft: StatefulTractogram, flip_ids: np.ndarray = None):
+    """Reverse streamlines, i.e. inverse the beginning and end
+
+    Params
```
Comment (ppoulin91) [inline dwi_ml/data/processing/streamlines/data_augmentation.py]: Parameters

File: dwi_ml/data/processing/streamlines/data_augmentation.py [hunk 2]
```diff
@@ -103,5 +103,33 @@ def cut_random_streamlines(
...
+    Returns
+    -------
+    new_sft: StatefulTractogram
+        Dipy object with reversed streamlines and data_per_point.
+    """
+    if flip_ids is None:
+        flip_ids = range(len(sft.streamlines))
+
+    new_streamlines = [s[::-1] if i in flip_ids else s for i, s in
+                       enumerate(sft.streamlines)]
```
Comment (AntoineTheb) [inline dwi_ml/data/processing/streamlines/data_augmentation.py]: Not sure if it's faster but

File: dwi_ml/data/processing/streamlines/data_augmentation.py [hunk 3]
```diff
@@ -103,5 +103,33 @@ def cut_random_streamlines(
...
+    new_data_per_point = [d[::-1] if i in flip_ids else d for i, d in
```
Comment (AntoineTheb) [inline dwi_ml/data/processing/streamlines/data_augmentation.py]: This should be tester before it's merged

File: dwi_ml/data/processing/streamlines/data_augmentation.py [hunk 4]
```diff
@@ -103,5 +105,36 @@ def cut_random_streamlines(
...
+        Dipy object with reversed streamlines and data_per_point.
+    """
+    if flip_ids is None:
+        flip_ids = range(len(sft.streamlines))
+
+    new_streamlines = [s[::-1] if i in flip_ids else s for i, s in
+                       enumerate(sft.streamlines)]
+    new_data_per_point = copy.deepcopy(sft.data_per_point)
+    for key in sft.data_per_point:
+        print('hey')
```
Comment (AntoineTheb) [inline dwi_ml/data/processing/streamlines/data_augmentation.py]: Hey !

Comment (EmmaRenauld) [reply dwi_ml/data/processing/streamlines/data_augmentation.py]: oups...

**Comment by AntoineTheb** [comment]
@EmmaRenauld Could you add "WIP" to the PR's title and  replace it with the appropriate "tag" (like ENH) once it's done ? Thanks !

**Comment by EmmaRenauld** [comment]
Ready now!

**Pull request merged**